### PR TITLE
#1-4Task. Paul lebedev

### PR DESCRIPTION
**Pull request by lenivecc**
https://travis-ci.org/github/lenivecc/js-assignments

### CODE REVIEW

File: task/01-strings-tasks.js [hunk 1]
```diff
@@ -201,7 +203,33 @@ function extractEmails(str) {
...
+            }
+            else {
+                if (i == 0) rectang += "│"; else
+                if (i == width - 1) rectang +="│\n";
+                else rectang +=" ";
+            }
+        }
+    }
+    return rectang;
 }
```
Comment (FedorovD) [inline task/01-strings-tasks.js:233]: `var` is deprecated. It has no sense to use for browsers code even currently.

File: task/02-numbers-tasks.js [hunk 2]
```diff
@@ -54,9 +54,10 @@ function getCicleCircumference(radius) {
  *  -3, 3  => 0
  */
 function getAverage(value1, value2) {
-    throw new Error('Not implemented');
+     if(((value1 + value2) / 2)<=1.7976931348623157e+308)
+     return ((value1+value2)/2);
+    else return 1.7976931348623157e+308;
 }
```
Comment (FedorovD) [inline task/02-numbers-tasks.js:60]: use Number.MAX_VALUE and Number.MIN_VALUE

File: task/02-numbers-tasks.js [hunk 3]
```diff
@@ -202,7 +220,12 @@ function roundToPowerOfTen(num, pow) {
...
  */
 function isPrime(n) {
-    throw new Error('Not implemented');
+    if(n<=3 || n==5 || n==7) return true;
+    else{
+    if (n%2!=0 && n%3!=0 && n%5!=0 && n%7!=0)
+    return true;
+    else return false;
+    }
 }
```
Comment (FedorovD) [inline task/02-numbers-tasks.js:229]: Please follow js code style
https://javascript.info/coding-style

File: task/04-arrays-tasks.js [hunk 4]
```diff
@@ -116,7 +121,11 @@ function removeFalsyValues(arr) {
  *    [ 'a', 'b', 'c', 'd', 'e', 'f', 'g' ]  => [ 'A', 'B', 'C', 'D', 'E', 'F', 'G' ]
  */
 function getUpperCaseStrings(arr) {
-   throw new Error('Not implemented');
+   let new_arr = arr.map((currentValue, index, array) => {
+      currentValue=array[index].toString().toUpperCase();
+      return currentValue;
+   });
+   return new_arr;
 }
```
Comment (FedorovD) [inline task/04-arrays-tasks.js:129]: extra steps in the code
Please, see the first example here https://www.w3schools.com/jsref/jsref_map.asp

File: task/04-arrays-tasks.js [hunk 5]
```diff
@@ -131,7 +140,11 @@ function getUpperCaseStrings(arr) {
  *    [ 'angular', 'react', 'ember' ] => [ 7, 5, 5 ]
  */
 function getStringsLength(arr) {
-   throw new Error('Not implemented');
+   let new_arr = arr.map((currentValue,index,array) => {
+      currentValue = array[index].length;
+      return currentValue;
+   });
+   return new_arr;
 }
```
Comment (FedorovD) [inline task/04-arrays-tasks.js:148]: same as above

File: task/04-arrays-tasks.js [hunk 6]
```diff
@@ -160,7 +174,9 @@ function insertItem(arr, item, index) {
  *    [ 'a', 'b', 'c', 'd'], 3  => [ 'a', 'b', 'c' ]
  */
 function getHead(arr, n) {
-   throw new Error('Not implemented');
+   return arr.filter(item => {
+      if(arr.indexOf(item) <n)
+   return item; });
 }
```
Comment (FedorovD) [inline task/04-arrays-tasks.js:180]: code style

**Comment by FedorovD** [comment]
Please correct code according to the comments and all code to common js code style.
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

**Comment by lenivecc** [comment]
All fixed, but i dont understand about last your comment. I return last row "return item; });" one line above...

**Comment by FedorovD** [comment]
> All fixed, but i dont understand about last your comment. I return last row "return item; });" one line above...

image


**Closing the PR as accepted! Thanks!**

**Pull request closed without merging**